src/coreclr/jit/morph.cpp

```diff
@@ -9727,48 +9727,75 @@ GenTreeHWIntrinsic* Compiler::fgOptimizeForMaskedIntrinsic(GenTreeHWIntrinsic* n
         return node;
     }
 #elif defined(TARGET_ARM64)
-    // TODO-SVE: This optimisation is too naive. It needs to calculate the full cost of the instruction
-    //           vs using the predicate version, taking into account all input arguements and all uses
-    //           of the result.
-    // return fgMorphTryUseAllMaskVariant(node);
+    return fgMorphTryUseAllMaskVariant(node);
 #else
 #error Unsupported platform
 #endif
     return nullptr;
 }
 
 #ifdef TARGET_ARM64
-//------------------------------------------------------------------------
-// canMorphVectorOperandToMask: Can this vector operand be converted to a
-//                              node with type TYP_MASK easily?
-//
-bool Compiler::canMorphVectorOperandToMask(GenTree* node)
-{
-    return varTypeIsMask(node) || node->OperIsConvertMaskToVector() || node->IsVectorZero();
-}
+
+// Conversion of mask to vector is one instruction.
+static constexpr const weight_t costOfConvertMaskToVector = 1.0;
+
+// Conversion of vector to mask is two instructions.
+static constexpr const weight_t costOfConvertVectorToMask = 2.0;
```

Comment:
add a comment on why `costOfConvertVectorToMask` is expensive than `costOfConvertMaskToVector`